### PR TITLE
PIM-10576: Fix product model mass edit acl check

### PR DESCRIPTION
**Pull request by cemorin**
**Description (for Contributor and Core Developer)**

Add an error in product and product model deletion when product model deletion is not granted

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Component/Product/Job/DeleteProductsAndProductModelsTasklet.php [hunk 1]
```diff
@@ -189,9 +172,19 @@ protected function doDelete(array $entities): void
         $products = $this->filterProducts($entities);
         $productModels = $this->filterProductModels($entities);
 
-        $deletedProductsCount = $this->countProductsToDelete($products, $productModels);
         $deletedProductModelsCount = $this->countProductModelsToDelete($productModels);
 
+        // When deletion for product models is not granted, we add a warning and reset the array so that there will be
+        // no deletion nor for their children
+        if (!$this->securityFacade->isGranted('pim_enrich_product_model_remove')) {
```
Comment (olivierpontier) [inline src/Akeneo/Pim/Enrichment/Component/Product/Job/DeleteProductsAndProductModelsTasklet.php:179]: should we also check the similar ACL for the products before the deletion?

**Review by olivierpontier — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Component/Product/Job/DeleteProductsAndProductModelsTasklet.php [hunk 2]
```diff
@@ -189,9 +172,19 @@ protected function doDelete(array $entities): void
...
+            $this->stepExecution->addWarning('Access forbidden. You are not allowed to delete product models', [], new DataInvalidItem($productModels));
+            $this->stepExecution->incrementSummaryInfo('skipped_deleted_product_models', $deletedProductModelsCount);
+            $deletedProductModelsCount = 0;
+            $productModels = [];
```
Comment (ghoussard) [inline src/Akeneo/Pim/Enrichment/Component/Product/Job/DeleteProductsAndProductModelsTasklet.php:183]: You can't early return ? It's less complex

```suggestion
            return 0;
```

**Review by ghoussard — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Component/Product/Job/DeleteProductsAndProductModelsTasklet.php (hunk 2 above)
Comment (cemorin) [reply src/Akeneo/Pim/Enrichment/Component/Product/Job/DeleteProductsAndProductModelsTasklet.php:183]: We still want to get through the following code, which handles the step execution informations and the product deletion.
But it's maybe not clear enough in the comment? (let me know)

File: src/Akeneo/Pim/Enrichment/Component/Product/Job/DeleteProductsAndProductModelsTasklet.php (hunk 1 above)
Comment (cemorin) [reply src/Akeneo/Pim/Enrichment/Component/Product/Job/DeleteProductsAndProductModelsTasklet.php:179]: For now the product deletion right is handled on the front side (the button delete is not available when the right is not granted) but we couldn't do the same for the product models.
But I can add a ticket for an improvement in our board :+1:

**Pull request merged**